backend/pkg/api/data_access/notifications.go

```diff
@@ -62,19 +69,234 @@ func (d *DataAccessService) GetNotificationSettings(ctx context.Context, userId
+	eventName = fmt.Sprintf("%s:%s", networkName, types.NetworkGasBelowThresholdEventName)
+	d.AddOrRemoveEvent(&eventsToInsert, &eventsToDelete, settings.IsGasBelowSubscribed, userId, eventName, "", epoch, settings.GasBelowThreshold.Div(decimal.NewFromInt(params.GWei)).InexactFloat64())
+	eventName = fmt.Sprintf("%s:%s", networkName, types.NetworkParticipationRateThresholdEventName)
+	d.AddOrRemoveEvent(&eventsToInsert, &eventsToDelete, settings.IsParticipationRateSubscribed, userId, eventName, "", epoch, settings.ParticipationRateThreshold)
+	eventName = fmt.Sprintf("%s:%s", networkName, types.RocketpoolNewClaimRoundStartedEventName)
+	d.AddOrRemoveEvent(&eventsToInsert, &eventsToDelete, settings.IsNewRewardRoundSubscribed, userId, eventName, "", epoch, 0)
+
+	// Insert all the events or update the threshold if they already exist
+	if len(eventsToInsert) > 0 {
+		insertDs := goqu.Dialect("postgres").
+			Insert("users_subscriptions").
+			Cols("user_id", "event_name", "event_filter", "created_ts", "created_epoch", "event_threshold").
+			Rows(eventsToInsert).
+			OnConflict(goqu.DoUpdate(
+				"user_id, event_name, event_filter",
+				goqu.Record{"event_threshold": goqu.L("EXCLUDED.event_threshold")},
+			))
+
+		query, args, err := insertDs.Prepared(true).ToSQL()
+		if err != nil {
+			return fmt.Errorf("error preparing query: %v", err)
+		}
+
+		_, err = tx.ExecContext(ctx, query, args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Delete all the events
+	if len(eventsToDelete) > 0 {
+		deleteDs := goqu.Dialect("postgres").
+			Delete("users_subscriptions").
+			Where(goqu.Or(eventsToDelete...))
+
+		query, args, err := deleteDs.Prepared(true).ToSQL()
+		if err != nil {
+			return fmt.Errorf("error preparing query: %v", err)
+		}
+
+		_, err = tx.ExecContext(ctx, query, args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return fmt.Errorf("error committing tx to update general notification settings: %w", err)
+	}
+	return nil
 }
 func (d *DataAccessService) UpdateNotificationSettingsPairedDevice(ctx context.Context, userId uint64, pairedDeviceId string, name string, IsNotificationsEnabled bool) error {
-	return d.dummy.UpdateNotificationSettingsPairedDevice(ctx, userId, pairedDeviceId, name, IsNotificationsEnabled)
+	_, err := d.userWriter.ExecContext(ctx, `
+		UPDATE users_devices 
+		SET 
+			device_name = $1,
+			notify_enabled = $2
+		WHERE user_id = $3 AND device_identifier = $4`,
+		name, IsNotificationsEnabled, userId, pairedDeviceId)
+	return err
 }
 func (d *DataAccessService) DeleteNotificationSettingsPairedDevice(ctx context.Context, userId uint64, pairedDeviceId string) error {
-	return d.dummy.DeleteNotificationSettingsPairedDevice(ctx, userId, pairedDeviceId)
+	_, err := d.userWriter.ExecContext(ctx, `
+		DELETE FROM users_devices 
+		WHERE user_id = $1 AND device_identifier = $2`,
+		userId, pairedDeviceId)
+	return err
 }
```

Comment:
Could consider returning an error here for now if no row was updated/deleted as currently there is no proper input parameter validation in api layer (see the TODO there), just for good measure

Reply:
Sure, I added it with a TODO that it could be deleted once the API layer has further checks.